examples/ImageRecon/GEOmetrics/utils.py

```diff
@@ -14,489 +14,485 @@
 
 import torch 
 from torchvision import transforms
-from torchvision.transforms import Normalize as norm 
-import trimesh
-from sklearn.preprocessing import normalize
 import kaolin as kal
-from PIL import Image
 from collections import defaultdict
 import numpy as np
-from kaolin.rep import TriangleMesh
-import kaolin as kal
-
+from torch._six import container_abcs
 
+from kaolin.rep import TriangleMesh
 
 
 preprocess = transforms.Compose([
-   transforms.Resize(224),
-   transforms.ToTensor()
+    transforms.Resize(224),
+    transforms.ToTensor()
 ])
 
 
-def collate_fn(data): 
-	new_data = {}
-	for k in data[0].keys():
-		
-		if k in ['points','norms', 'imgs', 'cam_mat', 'cam_pos']:
-			new_info = tuple(d[k] for d in data)
-			new_info = torch.stack(new_info, 0)
-		elif k in ['adj']: 
-			
-			adj_values = tuple(d[k].coalesce().values() for d in data)
-			adj_indices = tuple(d[k].coalesce().indices() for d in data)
-			new_data['adj_values'] = adj_values
-			new_data['adj_indices'] = adj_indices
-			
-		else: 
-			new_info = tuple(d[k] for d in data)
-
-		new_data[k] = new_info
-	return new_data
-
-
-
-
-
-def get_pooling_index( positions, cam_mat, cam_pos, dims):
-	device = positions.device
-	#project points into 2D
-	positions = positions * .57  # accounting for recaling in 3Dr2n
-	positions = positions - cam_pos 
-	positions = torch.mm(positions,cam_mat.permute(1,0))
-	positions_xs =  positions[:, 1] / positions[:, 2]
-	positions_ys = -positions[:, 0] / positions[:, 2] 
-
-	# do bilinear interpolation over pixel coordiantes
-	data_meta = defaultdict(list)
-
-	for dim in dims:
-		focal_length = 250./224. * dim
-		xs = positions_xs * focal_length + dim/2.
-		ys = positions_ys * focal_length + dim/2.
-
-		cur_xs = torch.clamp(xs , 0, dim - 1)
-		cur_ys = torch.clamp(ys , 0, dim - 1)
-
-
-		# img = np.zeros((dim,dim))
-		# for x,y in zip (cur_xs, cur_ys): 
-		# 	img[x.int(), y.int()] = 255
-		# from PIL import Image
-		# Image.fromarray(img).show()
-		
-		# exit()
-		
-		
-		x1s, y1s, x2s, y2s = torch.floor(cur_xs), torch.floor(cur_ys), torch.ceil(cur_xs), torch.ceil(cur_ys)
-
-		A = x2s - cur_xs
-		B = cur_xs - x1s
-		G = y2s - cur_ys
-		H = cur_ys - y1s
-
-		y1s = y1s + torch.arange(positions.shape[0]).float().to(device)*dim 
-		y2s = y2s + torch.arange(positions.shape[0]).float().to(device)*dim 
-
-		data_meta['A'].append(A.float().unsqueeze(0))
-		data_meta['B'].append(B.float().unsqueeze(0))
-		data_meta['G'].append(G.float().unsqueeze(0))
-		data_meta['H'].append(H.float().unsqueeze(0))
-		data_meta['x1s'].append(x1s.long().unsqueeze(0))
-		data_meta['x2s'].append(x2s.long().unsqueeze(0))
-		data_meta['y1s'].append(y1s.long().unsqueeze(0))
-		data_meta['y2s'].append(y2s.long().unsqueeze(0))
-
-	for key in data_meta:
-		data_meta[key] = torch.cat(data_meta[key], dim=0)
-	return data_meta
-
-
-
+def collate_fn(batch):
```

Comment:
I wonder if this shouldn't be part of our core library (or having collate method associated to each dataset class)

Reply:
I agree. Would also address issue #95. Let's put that work in a subsequent PR.